taskgraph/Task.py

```diff
@@ -473,16 +473,32 @@ def _task_executor(self):
             except queue.Empty:
                 # no tasks are waiting could be because the taskgraph is
                 # closed or because the queue is just empty.
-                if self._closed and not self._task_dependent_map:
-                    # the task graph is signaling executors to stop,
-                    # since the self._task_dependent_map is empty the
-                    # executor can terminate.
+                if (self._closed and len(self._completed_task_names) ==
+                        self._added_task_count):
+                    # the graph is closed and there are as many completed tasks
+                    # as there are added tasks, so none left. The executor can
+                    # terminate.
+                    self._executor_thread_count -= 1
+                    if self._executor_thread_count == 0 and self._worker_pool:
+                        # only the last executor should terminate the worker
+                        # pool, because otherwise who knows if it's still
+                        # executing anything
+                        try:
+                            self._worker_pool.close()
+                            self._worker_pool.terminate()
+                            self._worker_pool = None
+                        except Exception:
+                            # there's the possibility for a race condition here
+                            # where anothe thread already closed the worker

```

Comment:
Typo on `another`

Reply:
Fixed a couple more typos in 1d472d2635d0544bf3aa2fbad89d6dc1ef732eef